src/AppBundle/Controller/DashboardController.php

```diff
@@ -0,0 +1,344 @@
+            $chart->setLegend($component->getLegend());
+            $chart->setRequestSql($component->getRequestSQL());
+            $chart->setTypeGraph($component->getTypeGraph());
+            $chart->setXAxis($component->getXAxis());
+            $chart->setYAxis($component->getYAxis());
+            $chart = $chart->generateChart($key);
+            array_push($charts, $chart);
+            array_push($componentName, $component->getName());
+            array_push($componentId, $component->getId());
+            array_push($sizeComponent, $component->getSizeComponent());
+            $forms[$component->getId()] = $this->get('form.factory')->createNamedBuilder($component->getId(), new ComponentType(), $component)->getForm();
+        }
+        //.......Get current dashboard object
+        $em = $this->getDoctrine()->getManager();
+        $dashboard = $em->getRepository('AppBundle:Dashboard')
+            ->find($id);
+        //.......Set dashboard's creator
+        $creator = $dashboard->getCreator();
+
+        //.......Generate form for "Edit Dashboard"
+        $form = $this->get('form.factory')->create(new DashboardType(), $dashboard);
+        $form->handleRequest($request);
+
+        //.......Check if the method is POST and which form component is submitted
+        if ($request->isMethod('POST')) {
+
+            foreach ($forms as $key => $oneForm) {
+                $forms[$key]->handleRequest($request);
+            }
+            foreach ($forms as $key => $oneForm) {
+                if ($oneForm->isSubmitted() && $oneForm->isValid()) {
+
+                    $em = $this->getDoctrine()
+                        ->getManager();
+                    $component = $em->getRepository('AppBundle:Component')
+                        ->find($key);
+                    //.......Change graphic's type
+                    if ($oneForm->get('linechart')->isClicked()) {
+                        $component->setTypeGraph('linechart');
+                    }
+                    if ($oneForm->get('column')->isClicked()) {
+                        $component->setTypeGraph('column');
+                    }
+                    if ($oneForm->get('area')->isClicked()) {
+                        $component->setTypeGraph('area');
+                    }
+                    if ($oneForm->get('bar')->isClicked()) {
+                        $component->setTypeGraph('bar');
+                    }
+                    //.......Save changes made on component
+                    $em->persist($component);
+                    $em->flush();
+
+                    return $this->redirectToRoute('app_dashboard', array(
+                        'id' => $id,
+                        'component' => $component,
+                    ));
+                }
+            }
+        }
+
+        //.......If dashboard edit form was submitted, save changes to database
+        if ($form->isSubmitted() && $form->isValid()) {
```

Comment:
The form will always be valid because there are no validation rules.
You need to add them in your entities.
For example for the dashboard name you would add `@Assert\NotNull()` on the attribute.
Here's the [documentation](https://symfony.com/doc/current/validation.html).